builder/src/core/node.rs

```diff
@@ -310,17 +310,23 @@ impl Node {
             chunk.set_blob_index(blob_index);
             chunk.set_index(chunk_index);
             chunk.set_file_offset(file_offset);
+            let mut dumped_size = chunk.compressed_size();
```

Comment:
Seems better to leave `dumped_size` as it is. Because `dumped_size` have an initial value, `chunk.compressed_size()`, which is non-zero for zran-like referenced type. So `dumped_size` is non-empty and can't be wrapped by `Option`.